### DISCUSSION TITLE
Easy feature request - Integrate existing RGBD Video solution

### DISCUSSION THREAD

**Discussion by jeffscottward** [google/model-viewer#3085] [Ideas]
There is already a working example of 3D video at http://record3d.xyz with more info at https://record3d.app/features (check the source after uploading a sample mp4)

If you download the iOS app and pay for the pro + expansion to get RGBD video, you can record and export a sample to test with but I'm sure there's sample videos out there anyways.

The author says exporting to GLTF isn't really a true export and results in ginormous .bin files so I doubt AR is possible but the in browser rendering is already done. Would be nice to have it in a neat package here.

**Comment by elalish** [comment]
I'm not really sure how that  would fit into our product. Those don't look like they're really designed to be spun around and viewed from the side or back. What use case do you have in mind?

**Reply by jeffscottward** [reply]
It is only from one angle of course but 3D video is a category that falls under the general AR umbrella that differs from static model captures or virtual native content from DCC's.

My use case is a purchasable content platform that covers scans, filters, portals, 360 photos, static/animated models, interactive content, the whole spectrum basically.

.glb, .USDz, .real (Aero), .js (WebXR), .mp4 (rgbd)

It would be nice to just throw an rgbd .mp4 at it and it 'just works'.

There is no 1-line solution to this file format yet. 

If there is content that can standalone in its own file and follows a spec, I think it's worth integrating if it's minimal effort.

**Reply by elalish** [reply]
I'm not familiar with the RGBD spec; who has standardized it? If it's minimal effort, then please feel free to open a PR to add it.

**Reply by jeffscottward** [reply]
It's basically a Microsoft Kinect invention from back then.

I can open a PR but i don't know 3D software engineering :-/

Edit:
I can't find a canonical reference other than some sample datasets and tools to extract the data.

https://rgbd-dataset.cs.washington.edu/demos.html

Maybe this?

**Reply by elalish** [reply]
Yeah, that doesn't sound like a standard; just an engineering prototype. I don't think we'll be supporting that until it's much more widely adopted.
